visualization/app/codeCharta/ui/fileExtensionBar/fileExtensionBar.component.ts

```diff
@@ -27,6 +34,42 @@ export class FileExtensionBarController implements CodeMapPreRenderServiceSubscr
 		this.potentiallyAddNoneExtension()
 	}
 
+	public highlightBarHoveredBuildings(extension: string) {
+		const mapBuildings: CodeMapBuilding[] = this.threeSceneService
+			.getMapMesh()
+			.getMeshDescription()
+			.buildings.filter(building => building.node.isLeaf)
+		if (extension === "other") {
+			this.findBuildingsSummarizedInOtherDistribution(mapBuildings)
+		} else {
+			mapBuildings.forEach(building => {
+				if (FileExtensionCalculator.estimateFileExtension(building.node.name) === extension) {
+					this.threeSceneService.addBuildingToHighlightingList(building)
+				}
+			})
+		}
+
+		this.threeSceneService.highlightBuildings()
+	}
+	private findBuildingsSummarizedInOtherDistribution(mapBuildings: CodeMapBuilding[]) {
+		const visibleDistributionEndings: String[] = []
+		this._viewModel.distribution
+			.filter(metric => metric.fileExtension != "other")
+			.forEach(metricDestribution => {
+				visibleDistributionEndings.push(metricDestribution.fileExtension)
+			})
```

Comment:
I would refactor this to the following:

```
const visibleDistributionEndings: string[] = this._viewModel.distribution
	.filter(metric => metric.fileExtension != "other")
	.map(metric => metric.fileExtension)
```

Reply:
Yes, thats better and shorter! Thanks! :)